mmdeploy/codebase/mmdet3d/deploy/voxel_detection.py

```diff
@@ -50,107 +67,149 @@ def build_pytorch_model(self,
+        test_pipeline = Compose(test_pipeline)
+        box_type_3d, box_mode_3d = \
+            get_box_type(cfg.test_dataloader.dataset.box_type_3d)
+
+        data = []
+        data_ = dict(
+            lidar_points=dict(lidar_path=pcd),
+            timestamp=1,
+            # for ScanNet demo we need axis_align_matrix
+            axis_align_matrix=np.eye(4),
+            box_type_3d=box_type_3d,
+            box_mode_3d=box_mode_3d)
+        data_ = test_pipeline(data_)
+        data.append(data_)
+
+        collate_data = pseudo_collate(data)
+        data[0]['inputs']['points'] = data[0]['inputs']['points'].to(
+            self.device)
+
+        if preproc is not None:
+            collate_data = preproc(collate_data, False)
+            del collate_data['inputs']['voxels']['voxel_centers']
+        return collate_data, collate_data['inputs']
 
     def visualize(self,
+                  image: Union[str, np.ndarray],
                   model: torch.nn.Module,
-                  image: str,
                   result: list,
                   output_file: str,
-                  window_name: str,
+                  window_name: str = '',
                   show_result: bool = False,
-                  score_thr: float = 0.3):
-        """Visualize predictions of a model.
+                  draw_gt: bool = False,
+                  data_samples=None,
+                  **kwargs):
+        """_summary_
 
         Args:
-            model (nn.Module): Input model.
-            image (str): Pcd file to draw predictions on.
-            result (list): A list of predictions.
-            output_file (str): Output file to save result.
-            window_name (str): The name of visualization window. Defaults to
-                an empty string.
-            show_result (bool): Whether to show result in windows, defaults
-                to `False`.
-            score_thr (float): The score threshold to display the bbox.
-                Defaults to 0.3.
+            image (Union[str, np.ndarray]): _description_
+            result (list): _description_
+            output_file (str): _description_
+            window_name (str, optional): _description_. Defaults to ''.
+            show_result (bool, optional): _description_. Defaults to False.
+            draw_gt (bool, optional): _description_. Defaults to False.
         """
-        from mmdet3d.apis import show_result_meshlab
-        data = VoxelDetection.read_pcd_file(image, self.model_cfg, self.device)
-        show_result_meshlab(
-            data,
-            result,
-            output_file,
-            score_thr,
+        cfg = self.model_cfg
+        visualizer = super().get_visualizer(window_name, output_file)
+        visualizer.dataset_meta = _get_dataset_metainfo(cfg)
+
+        # show the results
+        _, data_input = self.create_input(pcd=image)
+
+        if Backend(window_name) != Backend.PYTORCH:
```

Comment:
better to use `get_backend` to get backend type from model config.
```from mmdeploy.utils import get_backend```

Reply:
done.